File: SimpleHabit/app/src/main/java/com/me/simplehabit/data/models/CurrentProgramModel.java
package com.me.simplehabit.data.models;

import com.me.simplehabit.data.vos.CurrentProgramVO;

public interface CurrentProgramModel {

    CurrentProgramVO getCurrentProgram(CurrentProgramDelegate delegate,boolean isForce);

    CurrentProgramVO getCurrentProgram();

    interface CurrentProgramDelegate{

        void onCurrentProgramFetchFromNetwork(CurrentProgramVO currentProgramVO);

        void onErrorOnProgramFetch(String message);
    }
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/data/models/CurrentProgramModel.kt
package com.me.simplehabit.data.models

import androidx.lifecycle.LiveData
import com.me.simplehabit.data.vos.CurrentProgramVO

interface CurrentProgramModel {

    fun getCurrentProgram(): LiveData<CurrentProgramVO>

    fun loadCurrentProgram(isForce: Boolean, networkFailure: (String) -> Unit): LiveData<CurrentProgramVO>

}
